jax/interpreters/ad.py

```diff
@@ -109,12 +109,16 @@ def vjp(traceable, primals, has_aux=False):
     out_primals, pvals, jaxpr, consts = linearize(traceable, *primals)
   else:
     out_primals, pvals, jaxpr, consts, aux = linearize(traceable, *primals, has_aux=True)
-  def vjp_(*cts):
+
+  def unbound_vjp(pvals, jaxpr, consts, *cts):
     cts = tuple(map(ignore_consts, cts, pvals))
     dummy_args = [UndefinedPrimal(v.aval) for v in jaxpr.invars]
     arg_cts = backward_pass(jaxpr, consts, dummy_args, cts)
     return map(instantiate_zeros, arg_cts)
 
+  # Ensure that vjp_ is a PyTree so that we can pass it from the forward to the backward
+  # pass in a custom VJP.
+  vjp_ =  Partial(partial(unbound_vjp, pvals, jaxpr), consts)
```

Comment:
Note to future self: I think this is safe because, while `pvals` can contain tracers in general in their consts, here I don't think they can (because they're the output tangent pvals from linearization) and in any case `ignore_consts` just drops the consts on the floor.

Reply:
If it makes you feel more comfortable, we could
```
assert all(pval[0] is not None for pval in pvals)
```
?